### PR TITLE
Ability to decide, at compile time, which "plrust-trusted-pgx" crate user functions should use

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**
If the `PLRUST_TRUSTED_PGX_OVERRIDE` environment variable exists when compiling **plrust** itself, we compile-in its value to be used as the `pgx = ` dependency for all generated user crates.

Example usage might be:

```shell
$ cd plrust
$ PLRUST_TRUSTED_PGX_OVERRIDE="pgx = { path = '~/code/plrust/plrust-trusted-pgx', package='plrust-trusted-pgx' }" \
cargo pgx run
 ```

This is really targeted for us to use in CI towards the goal of continuously testing plrust against `pgx:develop`.

### CODE REVIEW

**Comment by BradyBonnette** [comment]
Curious to know if there are any security implications with this? I suppose if someone is able to access and modify environment variables it's Not Great™️but wasn't sure if there was any other impacts..

**Comment by thomcc** [comment]
> Curious to know if there are any security implications with this? I suppose if someone is able to access and modify environment variables it's Not Great™️but wasn't sure if there was any other impacts..

An attacker that can control the environment can set LD_PRELOAD/DYLD_INSERT_LIBRARIES and pwn the process that way.

**Review by thomcc — APPROVED** [review]
Might be worth mentioning in https://github.com/tcdi/plrust/blob/main/doc/src/config-env-var.md?

**Comment by BradyBonnette** [comment]
> An attacker that can control the environment can set LD_PRELOAD/DYLD_INSERT_LIBRARIES and pwn the process that way.

Yeah I figured there were worse things that could happen.

**Review by BradyBonnette — APPROVED** [review]

**Review by thomcc — APPROVED** [review]
lgtm

**Comment by eeeebbbbrrrr** [comment]
Before I merge this  I'm going to attempt to go ahead and make the existing CI use this, so we're at least testing with the `plrust-trusted-pgx` crate that's in this repo instead of the one that's out on crates.io.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
       WORK_DIR: /home/ec2-user/actions-runner/_work/plrust/plrust
       ARTIFACT_USER_AWS_PROFILE: artifact-user
       AWS_CACHE_BUCKET: tcdi-ci-plrust-build-cache.private
+      PLRUST_TRUSTED_PGX_OVERRIDE: "pgx xxx= { path = '/home/ec2-user/actions-runner/_work/plrust/plrust/plrust-trusted-pgx', package='plrust-trusted-pgx' }"
```
[the remark this answers is not in the document]
Comment (BradyBonnette) [reply .github/workflows/ci.yml:33]: I can strong-arm it into submission. It's not that they "don't" work, it's that they "don't work like Github-hosted runners"

[the remark this answers is not in the document]
Comment (eeeebbbbrrrr) [reply .github/workflows/ci.yml]: As we discussed offline, I mistyped that on purpose so I could cause CI to fail -- wanted to make sure it was actually picking up these environment variables from where I thought.  [6a19a7f](https://github.com/tcdi/plrust/pull/276/commits/6a19a7f597b800b7e51559de7f232b50a8d3591e) fixes it.  waiting out that workflow run now...

**Pull request merged**